### PR TITLE
Fonts changing ingame

### LINKED ISSUE #23863: Feature request - Zoom for text
Game version: experimental 7426

Operating system: Win 10

Tiles or curses: Tiles

Mods active: All of them that are bundled; except Dino mod, generic guns, and the balancing mods except for the default (filthy clothes etc.)

### Feature Request

While 'Fullscreen' has the option to scale everything, windowed does not. On my screen that makes windowed impossible to read; the text is too small, so I have to use fullscreen. That wouldn't be a problem except fullscreen CTD's if I switch to another program or my desktop.

While fixing the CTD would resolve my own problems, having a zoom function for windowed format would be a nice feature in and of itself.

### PR DESCRIPTION
**Pull request by Night-Pryanik**
#### Summary
```SUMMARY: Interface "Added possibility to change font size settings in game options."```  

#### Purpose of change
Closes #23863.

#### Describe the solution
Added font width, height, size and blending options. This data is no longer read from `fonts.json`, but instead from `options.json`. Font typefaces are still in `fonts.json`.

#### Additional context
[CR] is because I need sensible min and max values for font settings. For now they are set as 8 and 100 respectfully.

default

### CODE REVIEW

File: src/options.cpp [hunk 1]
```diff
@@ -1476,6 +1476,58 @@ void options_manager::add_options_graphics()
 
     mOptionsSort["graphics"]++;
 
+    add( "FONT_BLENDING", "graphics", translate_marker( "Font blending" ),
+        translate_marker( "If true, fonts will look better." ),
```
Comment (Night-Pryanik) [inline src/options.cpp]: Please someone tell me what font blending is doing, so I can write a accurate description.

Comment (ifreund) [reply src/options.cpp]: Looks like it's making it look much nicer by using alpha values to create a smoother look. https://www.libsdl.org/projects/docs/SDL_ttf/SDL_ttf_35.html

**Comment by AMurkin** [comment]
Nice.
They all should depend on `Use tiles` I suppose.

**Comment by Night-Pryanik** [comment]
> They all should depend on `Use tiles` I suppose.

Why? Font size can be changed both on curses and tiles versions.

**Comment by AMurkin** [comment]
Well, yes. Not that option, but only for Tiles builds.
Found it: COPT_CURSES_HIDE.

**Comment by Night-Pryanik** [comment]
I understand, but again, why this option should be hidden for curses build?

**Comment by AMurkin** [comment]
Because the terminal application defines font (one font) settings?

**Comment by Night-Pryanik** [comment]
You mean in curses build there is only one type of font that defines text font, map font and overmap font?

**Comment by AMurkin** [comment]
I hope that someone who knows how it works and can explain it will comment.

**Comment by ralreegorganon** [comment]
Yes, a curses build just uses the terminal font. 

Just for demo purposes, I made the font Papyrus:

screen shot 2019-02-07 at 09 19 13
screen shot 2019-02-07 at 09 19 25

**Comment by Night-Pryanik** [comment]
Okay, I got it, thanks everyone for comments and explanations. I will add corresponding option soon.

**Comment by Night-Pryanik** [comment]
No, wait, I still have a couple of questions. I only moved options to change fonts' width, height and size, not the type of font used (this is still is `fonts.json`). I assume it is possible to change the fonts' width, height and size in curses build? And if so, then why restrict this possibility only to sdl builds?

**Comment by kevingranade** [comment]
I am not aware of a way to change the terminal font settings from the application.  Even if it's possible, we don't have support for it right now, so we should hide the options on curses builds.

**Comment by Treah** [comment]
Why do fonts in CDDA have 3 parameters when just about every other software only has a point size? Is it white space being used between the char or something? This has always kinda confused me. Also the built in fonts for CDDA look horrendous when you try scaling them up, its one of the reasons I import others manually. .

**Comment by ZhilkinSerg** [comment]
>  Also the built in fonts for CDDA look horrendous when you try scaling them up

These are raster fonts.

**Pull request merged**